logging/logrus/logger.go

```diff
@@ -76,12 +76,19 @@ type Logger struct {
 //
 func NewLogger(name string) *Logger {
 	logger := &Logger{
-		std:    lg.New(),
-		depth:  2,
-		tagmap: make(map[uint64]string, 64),
-		name:   name,
+		std:   lg.New(),
+		depth: 2,
+		name:  name,
 	}
 
+	// init tagmap
+	tags := make(map[uint64]string, 64)
+	logger.tagmap.Store(tags)
+
+	// init static fields
+	statics := make(map[string]interface{})
+	logger.staticFields.Store(statics)
```

Comment:
It seems that we will need mutex for maps. At least each map can have different mutex.